### PR TITLE
authentication, settings

### PR DESCRIPTION
**Pull request by sotomskir**

### CODE REVIEW

File: src/app/shared/nav/nav.component.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
                [mode]="(isHandset$ | async) ? 'over' : 'side'"
                [opened]="(isHandset$ | async) === false">
     <mat-toolbar>{{nav.title}}</mat-toolbar>
-    <mat-nav-list>
+    <mat-nav-list *ngxPermissionsExcept="'GUEST'">
```
Comment (bwojtyla) [inline src/app/shared/nav/nav.component.html]: enum

File: src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  get token() {
+    if (!this.decodedToken) {
+      this.loadToken();
+    }
+    return this.decodedToken;
+  }
+
+  get permissions(): string[] {
+    if (!this.token) {
+      return ['GUEST'];
```
Comment (bwojtyla) [inline src/app/core/services/auth.service.ts]: you should create enum for user type

File: src/app/shared/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  form: FormGroup;
+
+  constructor(private fb: FormBuilder, private authService: AuthService) {
+    this.form = fb.group({
```
Comment (bwojtyla) [inline src/app/shared/login/login.component.ts]: you should create form in "formclass" and add some validations

Comment (sotomskir) [reply src/app/shared/login/login.component.ts]: I agree with validations but why "formclass"? What's wrong with form in component?

File: src/app/shared/nav/nav.component.html (hunk 1 above)
Comment (sotomskir) [reply src/app/shared/nav/nav.component.html]: Done

File: src/app/core/services/auth.service.ts (hunk 2 above)
Comment (sotomskir) [reply src/app/core/services/auth.service.ts]: Done

File: src/app/shared/login/login.component.ts (hunk 3 above)
Comment (sotomskir) [reply src/app/shared/login/login.component.ts]: Done

**Pull request merged**